lib/versions.nix

```diff
@@ -1,13 +1,29 @@
 /* Version string functions. */
 { lib }:
 
-rec {
+with lib;
+let
+  truncate = n: v: concatStringsSep "." (take n (splitVersion v));
+  opTruncate = op: v0: v: let n = length (splitVersion v0); in
+     op (truncate n v) (truncate n v0);
+in rec {
 
   /* Break a version string into its component parts.
 
      Example:
        splitVersion "1.2.3"
        => ["1" "2" "3"]
+
+     Remark: the builtins version of `splitVersion` -- which is
```

Comment:
I'm wondering if this isn't a Frenchism. I would use `Note:` instead.

Reply:
What would be the problem with "remark", except that it is an understatement. "Disclaimer" would be more appropriate IMO
...